cloudfour/cloudfour.com-patterns
# Basic Button pattern

## Overview

Adds an `o-button` CSS/Twig object, with hover, active, focus and disabled states. Also includes `--secondary` and `--tertiary` modifiers.

I added [the focus-visible polyfill](https://github.com/WICG/focus-visible) because I enjoy the freedom it gives me to design _big_ focus states without fear of distracting mouse/touch users.

I decided to rename `examples` directories to `demo` because I got tired of typing "examples" over and over.

This might feel a little premature to tackle, but I wanted to start writing fundamental patterns soon so we can review and scare out any issues with our approach sooner rather than later. So please feel free to nitpick this PR!

### Known Issues

- This does not include buttons optimized for dark backgrounds. I started adding these but hit a bit of a rabbit hole getting a documentation page to show some previews with dark backgrounds, which made me think I should stop and make sure what I have so far is decent.
- It also does not include color variations like purple, pink, green, red, etc. It felt a little premature to do that right now.
- I think there will be an opportunity to share certain sizes or preferences with other form patterns, but I felt I would be _guessing_ at what form that should take until we started making those.

## Screenshots

![Screenshot_2020-03-04 Objects Button - Elements ⋅ Storybook](https://user-images.githubusercontent.com/69633/75927273-e6101700-5e20-11ea-8b3c-a1bad636c8bd.png)

## Testing

1. View deploy preview
1. Confirm focus state displays via keyboard but not via mouse/touch
1. Play with knobs in "canvas" view, confirm they work as expected
1. Confirm accessibility tab in "canvas" view passes

---

- Fixes #475
